src/EcomDestinationWriter.cs

```diff
@@ -1686,14 +1709,120 @@ private void WriteDetails(Dictionary<string, object> row, Dictionary<string, Col
     private void WritePrices(Dictionary<string, object> row, Dictionary<string, ColumnMapping> columnMappings, DataRow dataRow)
     {
         ColumnMapping priceIdColumn = null;
-        if (!columnMappings.TryGetValue("PriceID", out priceIdColumn) || string.IsNullOrEmpty(Converter.ToString(row[priceIdColumn.SourceColumn.Name])))
+        if (!columnMappings.TryGetValue("PriceID", out priceIdColumn) || string.IsNullOrEmpty(GetMergedValue(priceIdColumn, row)))
         {
             LastPriceId = LastPriceId + 1;
             dataRow["PriceID"] = "ImportedPRICE" + LastPriceId;
         }
-        if (!columnMappings.ContainsKey("PriceCurrency"))
+
+        if (!columnMappings.TryGetValue("PriceCurrency", out var priceCurrencyColumn))
+        {
+            dataRow["PriceCurrency"] = Ecommerce.Services.Currencies.GetDefaultCurrency().Code;
+        }
+        else
+        {
+            var priceCurrencyValue = GetMergedValue(priceCurrencyColumn, row);
+            if (string.IsNullOrWhiteSpace(priceCurrencyValue))
+            {
+                row[priceCurrencyColumn.SourceColumn.Name] = Ecommerce.Services.Currencies.GetDefaultCurrency().Code;
+            }
+        }
+
+        if (columnMappings.TryGetValue("PriceUserId", out var priceAccessUserColumn))
+        {
+            var userIdLookupValue = GetMergedValue(priceAccessUserColumn, row);
+            var userIDs = ExistingUsers.Select("AccessUserID='" + userIdLookupValue + "'").Select(r => r["AccessUserID"].ToString()).ToList();
+            if (userIDs.Count == 0)
+            {
+                userIDs = ExistingUsers.Select("AccessUserExternalID='" + userIdLookupValue + "'").Select(r => r["AccessUserID"].ToString()).ToList();
+            }
+
+            if (userIDs.Count > 0)
+            {
+                dataRow["PriceUserId"] = userIDs[0];
+                row["PriceUserId"] = userIDs[0];
+                if (userIDs.Count > 1)
+                {
+                    logger.Warn($"Found multiple Users with AccessUserExternalId={userIdLookupValue} using the first one with userId = {userIDs[0]}.");
+                }
+            }
+            else
+            {
+                logger.Warn($"Could not find any User with {userIdLookupValue} as User ID or ExternalId.");
+            }
+        }
+
+        if (columnMappings.TryGetValue("PriceUserGroupId", out var priceAccessUserGroupColumn))
+        {
+            var userGroupIdLookupValue = GetMergedValue(priceAccessUserGroupColumn, row);
+            var userIDs = ExistingUsers.Select("AccessUserExternalID='" + userGroupIdLookupValue + "'").Select(r => r["AccessUserID"].ToString()).ToList();
```

Comment:
If it is a real UserGroup then it has a type of: AccessUserType IN (2,10,11)
Currently this search will return not only the groups but also a users